pgtricks/pg_dump_splitsort_py3.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+import functools
+
+COPY_RE = re.compile(r'COPY .*? \(.*?\) FROM stdin;\n$')
+
+
+def try_float(s):
+    if not s or s[0] not in '0123456789.-':
+        # optimization
+        return s
+    try:
+        return float(s)
+    except ValueError:
+        return s
+
+
+def linecomp(l1, l2):
+    p1 = l1.split('\t', 1)
+    p2 = l2.split('\t', 1)
+    v1 = try_float(p1[0])
+    v2 = try_float(p2[0])
+    result = (v1 > v2) - (v1 < v2)
```

Comment:
cmp was removed in Python 3, I opted for the quick trick from [here](https://docs.python.org/3.0/whatsnew/3.0.html#ordering-comparisons)

Reply:
These comparisons fail for `VARCHAR` and `TEXT` fields with content that sometimes looks like an int or a float.

I solved this in #7 by converting both values to floats in one go and keeping as strings if either fails.